express-notes-api/index.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const notesObject = require('./data.json');
+const app = express();
+const fs = require('fs');
+let nextID = 1;
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const array = Object.keys(notesObject.notes).map(key => {
+    return notesObject.notes[key];
+  });
+  res.status(200);
+  res.json(array);
+}
+);
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!Number.isInteger(id) || id <= 0) {
+    res.status(400).json({ Error: 'id must be a postive integer' });
+  } else if (!notesObject.notes[id]) {
+    res.status(404).json({ Error: `cannot find id ${id}` });
+  } else {
+    res.status(200).json(notesObject.notes[req.params.id]);
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  const body = req.body;
+  if (!body.content) {
+    res.status(400).json({ Error: 'content is a required field' });
+  } else {
+    body.id = nextID;
+    nextID++;
+    notesObject.notes[body.id] = body;
+    fs.writeFile('./data.json', JSON.stringify(notesObject, null, 2) + '\n', err => {
+      if (err) {
+        res.status(500).json({ Error: 'An unexpected error occured' });
+      }
+    });
+    res.status(201).json(notesObject.notes[body.id]);
```

Comment:
Be sure to put your success response in your writeFile callback so that it doesn't get sent before the note is saved. you'll also need an `else` in the callback so that it's impossible for your server to attempt to respond more than once to the same request.